File: fitness-server/src/main/java/com/fitness/server/entity/WorkoutSet.java
package com.fitness.server.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

@TableName("workout_sets")
public class WorkoutSet {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long workoutExerciseId;
    private Long localId;
    private Integer setNumber;
    private Double weight;
    private Integer reps;
    private Boolean isCompleted;
    private Integer restTime;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getWorkoutExerciseId() { return workoutExerciseId; }
    public void setWorkoutExerciseId(Long workoutExerciseId) { this.workoutExerciseId = workoutExerciseId; }
    public Long getLocalId() { return localId; }
    public void setLocalId(Long localId) { this.localId = localId; }
    public Integer getSetNumber() { return setNumber; }
    public void setSetNumber(Integer setNumber) { this.setNumber = setNumber; }
    public Double getWeight() { return weight; }
    public void setWeight(Double weight) { this.weight = weight; }
    public Integer getReps() { return reps; }
    public void setReps(Integer reps) { this.reps = reps; }
    public Boolean getIsCompleted() { return isCompleted; }
    public void setIsCompleted(Boolean isCompleted) { this.isCompleted = isCompleted; }
    public Integer getRestTime() { return restTime; }
    public void setRestTime(Integer restTime) { this.restTime = restTime; }
}

File: app/src/main/java/com/fitness/training/data/entity/WorkoutSet.kt
package com.fitness.training.data.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "workout_sets",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutExercise::class,
            parentColumns = ["id"],
            childColumns = ["workoutExerciseId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class WorkoutSet(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val workoutExerciseId: Long,
    val setNumber: Int, // 组号
    val weight: Double, // 重量
    val reps: Int, // 次数
    val isCompleted: Boolean = true, // 是否完成
    val restTime: Int = 0 // 休息时间（秒）
)



